src/postamble.js

```diff
@@ -167,7 +167,18 @@ function callMain(args) {
     // that if we get here main returned zero.
     var ret = 0;
 #else
+#if ASYNCIFY == 2
+    {{{ runtimeKeepalivePush(); }}}
+
+#endif // ASYNCIFY == 2
+
     var ret = entryFunction(argc, argv);
+
+#if ASYNCIFY == 2
+    ret.finally((result) => {
+      {{{ runtimeKeepalivePop(); }}}
+    });
+#endif // ASYNCIFY == 2
```

Comment:
Maybe add comments here? I'm confused as to why this is needed.

Reply:
I added a comment, but in the future we could potentially get rid of runtimekeepalive stuff altogether for asyncify==2 since we now have a promise that will track when execution is done.

Reply:
I should also note this `callMain()` is getting a bit weird since we're mixing in async/sync and they don't really work the same e.g. (exception handling). This could be a refactor in the future.